ui/lib/apps/ClusterInfo/components/StoreLocationTree/index.tsx

```diff
@@ -9,12 +9,197 @@ import { Space } from 'antd'
 import { cyan, magenta, grey } from '@ant-design/colors'
 import { useTranslation } from 'react-i18next'
 
+import { TopologyStoreLocation } from '@lib/client'
+
+import styles from './index.module.less'
+
+//////////////////////////////////////
+
+type ShortStrMap = Record<string, string>
+
+export function getShortStrMap(
+  data: TopologyStoreLocation | undefined
+): ShortStrMap {
+  let allShortStrMap: ShortStrMap = {}
+
+  if (data === undefined) {
+    return allShortStrMap
+  }
+
+  // location labels
+  // failure-domain.beta.kubernetes.io/region => region
+  data.location_labels?.forEach((label) => {
+    if (label.indexOf('/') >= 0) {
+      const shortStr = label.split('/').pop()
+      if (shortStr) {
+        allShortStrMap[label] = shortStr
+      }
+    }
+  })
+
+  // location labels value
+  data.location_labels?.forEach((label) => {
+    // get label values
+    const labelValues: string[] = []
+    data.stores?.forEach((store) => {
+      const val = store.labels?.[label]
+      if (val) {
+        labelValues.push(val)
+      }
+    })
+    const shortStrMap = getArrShortStrMap(labelValues)
+    allShortStrMap = Object.assign(allShortStrMap, shortStrMap)
+  })
+
+  // tikv & tiflash nodes address
+  const addresses = (data.stores || []).map((s) => s.address!)
+  addresses.forEach((addr) => {
+    if (addr.startsWith('db-')) {
+      const shortStr = addr.split('.').shift()
+      if (shortStr) {
+        allShortStrMap[addr] = shortStr
+      }
+    }
+  })
+
+  return allShortStrMap
+}
+
+// input: ['aaa-111a.abc.123', 'aaa-222a.abc.123', 'aaa-333a.abc.123'], items in the array have either the same prefix or suffix, or both.
+// output:
+// {
+//   "aaa-111a.abc.123":"111a",
+//   "aaa-222a.abc.123":"222a",
+//   "aaa-333a.abc.123":"333a"
+// }
+export function getArrShortStrMap(strArr: string[]): ShortStrMap {
+  const shortStrMap: ShortStrMap = {}
+  const strSet = new Set(strArr)
+  if (strSet.size < 2) {
```

Comment:
Is this mean that there is no need to compress info when there is only one node.

Reply:
yep, you can't calculate the duplicated prefix or suffix for only one item.

Reply:
Got it when I read the code behind. Maybe the function name can simply describe the algorithm, or the algorithm can have its own function.

Reply:
how about `trimDuplicate`? do you have any better name?

Reply:
I think `trimDuplicate` is good! My imagination is poor. :D